### PR TITLE
Use node 16.x in tests and development

### PR DESCRIPTION
**Pull request by ajacksified**
This PR updates the .nvmrc to use Node 16, updates the CI / CD workflows to include 16 in the test matrix and for builds, and  rebuilds the package-lock.json using NPM 8.

### CODE REVIEW

File: .github/workflows/e2e-node.yml [hunk 1]
```diff
@@ -12,10 +12,10 @@ jobs:
     if: github.actor != 'dependabot[bot]' && ${{github.event.deployment_status.state == 'success'}}
     steps:
     - uses: actions/checkout@v2.3.4
-    - name: Use Node.js 14.x
+    - name: Use Node.js 16.x
       uses: actions/setup-node@v2.1.5
       with:
-        node-version: 14.x
+        node-version: 16.x
```
Comment (ThisIsMissEm) [inline .github/workflows/e2e-node.yml:18]: Should this be setup as a matrix against all supported node versions? 12.x, 14.x, 16.x?

**Review by ThisIsMissEm — APPROVED** [review]

File: .github/workflows/e2e-node.yml (hunk 1 above)
Comment (NSeydoux) [reply .github/workflows/e2e-node.yml:18]: Should the e2e tests run across multiple node versions, or is that only required for unit tests ?

Comment (ajacksified) [reply .github/workflows/e2e-node.yml:18]: imo, we only need the matrix on unit tests, which are very likely to catch inconsistencies across node versions. Past that (given that we have 100% unit coverage), there are very quickly diminishing returns.

**Comment by ajacksified** [comment]
@ThisIsMissEm Merging since you approved, but feel free to re-open the conversation about node matrices in e2e tests.

**Pull request merged**